x-pack/plugins/files/public/components/image/viewport_observer.ts

```diff
@@ -25,7 +25,10 @@ export class ViewportObserver {
       opts: IntersectionObserverInit
     ) => IntersectionObserver
   ) {
-    this.intersectionObserver = getIntersectionObserver(this.handleChange, { root: null });
+    this.intersectionObserver = getIntersectionObserver(this.handleChange, {
```

Comment:
As discussed, if small tweaks don't fix it, maybe we could try some open source React utilities for element in-viewport detection.